sdk/python/kfp/components/_component_store.py

```diff
@@ -19,9 +19,10 @@
 
 
 class ComponentStore:
-    def __init__(self, local_search_paths=None, url_search_prefixes=None):
+    def __init__(self, local_search_paths=None, url_search_prefixes=None, auth=None):
```

Comment:
it's not very clear to me what type auth is, is it an auth token string?

Reply:
It's an auth object for the requests library. See https://requests.readthedocs.io/en/master/user/authentication/
They accept severla different kinds of authentication including `(user, password)` pairs, some predefined methods and custom functions.